helm/hpcc/templates/eclagent.yaml

```diff
@@ -50,20 +50,22 @@ data:
 {{ include "hpcc.addConfigMapVolumeMount" .me | indent 12 }}
 {{ include "hpcc.addDataVolumeMount" . | indent 12 }}
 {{ include "hpcc.addDllVolumeMount" .root | indent 12 }}
-{{ include "hpcc.addSecretVolumeMounts" (dict "root" .root "categories" (list "system" "ecl" "storage" ) ) | indent 12 }}
+{{ include "hpcc.addSecretVolumeMounts" (dict "root" .root "categories" .secretsCategoriesSystem ) | indent 12 }}
           volumes:
 {{ include "hpcc.addConfigMapVolume" .me | indent 10 }}
 {{ include "hpcc.addDataVolume" . | indent 10 }}
 {{ include "hpcc.addDllVolume" .root | indent 10 }}
-{{ include "hpcc.addSecretVolumes" (dict "root" .root "categories" (list "system" "ecl" "storage" ) ) | indent 10 }}
+{{ include "hpcc.addSecretVolumes" (dict "root" .root "categories" .secretsCategoriesSystem ) | indent 10 }}
           restartPolicy: Never
       backoffLimit: 0
 {{- end }}
 {{- end -}}
 
 {{ range $.Values.eclagent -}}
 {{- if not .disabled -}}
-{{- $commonCtx := dict "root" $ "me" . }}
+{{- $secretsCategoriesUser := list "ecl-user" "ecl" "storage" }}
```

Comment:
If these lists are different then the names should reflect why they are different e.g. Agent/Job rather than the items that they contain.  From the names I couldn't tell what distinguished them.